cairo_programs/math_cmp_and_pow_integration_tests.cairo

```diff
@@ -0,0 +1,108 @@
+%builtins range_check
+
+from starkware.cairo.common.alloc import alloc
+from starkware.cairo.common.pow import pow
+from starkware.cairo.common.math_cmp import (
+    is_not_zero,
+    is_nn,
+    is_le,
+    is_nn_le,
+    is_in_range,
+    is_le_felt,
+)
+
+const CONSTANT = 3 ** 10
+
+func fill_array_with_pow{range_check_ptr}(array_start: felt*, base: felt, step: felt, exp: felt, iter: felt, last: felt) -> ():
+    if iter == last:
+        return()
+    end
+    let (res) = pow(base + step ,exp)
+    assert array_start[iter] = res
+    return fill_array_with_pow(array_start, base + step, step, exp, iter + 1, last)
+end
+
+func test_is_not_zero{range_check_ptr}(base_array: felt*, new_array: felt*, iter: felt, last: felt) -> ():
+    if iter == last:
+        return()
+    end
+    let (res) = is_not_zero(base_array[iter])
+    assert new_array[iter] = res
+    return test_is_not_zero(base_array, new_array, iter + 1, last)
+end
+
+func test_is_nn{range_check_ptr}(base_array: felt*, new_array: felt*, iter: felt, last: felt) -> ():
+    if iter == last:
+        return()
+    end
+    let (res) = is_nn(base_array[iter])
+    assert new_array[iter] = res
+    return test_is_nn(base_array, new_array, iter + 1, last)
+end
+
+func test_is_le{range_check_ptr}(base_array: felt*, new_array: felt*, constant: felt ,iter: felt, last: felt) -> ():
+    if iter == last:
+        return()
+    end
+    let (res) = is_le(base_array[iter], constant )
+    assert new_array[iter] = res
+    return test_is_le(base_array, new_array, constant, iter + 1, last)
+end
+
+func test_is_nn_le{range_check_ptr}(base_array: felt*, new_array: felt*, constant: felt ,iter: felt, last: felt) -> ():
```

Comment:
Is constant needed as a variable?, The functions could use the constant value directly.